shared/chat/conversation/input-area/user-mention-hud/mention-hud-container.js

```diff
@@ -12,27 +14,57 @@ const mapStateToProps = (state: TypedState, {filter, conversationIDKey}) => {
   }
 }
 
+const mapDispatchToProps = (dispatch: Dispatch) => ({
+  _loadParticipants: conversationIDKey =>
+    dispatch(Chat2Gen.createMetaRequestTrusted({conversationIDKeys: [conversationIDKey], force: true})),
+})
+
 const mergeProps = (stateProps, dispatchProps, ownProps) => {
   const meta = stateProps._metaMap.get(stateProps.conversationIDKey)
   let participants = meta ? meta.participants : I.Set()
+  let _generalChannelConversationIDKey = ''
   // Get the general channel participants instead
   if (meta && meta.teamType === 'big' && meta.channelname !== 'general') {
     const m = stateProps._metaMap.find(m => m.teamname === meta.teamname && m.channelname === 'general')
     if (m) {
       participants = m.participants
+      _generalChannelConversationIDKey = m.conversationIDKey
+    } else {
+      logger.info(`MentionHUD: no meta found for general channel in team ${meta.teamname}`)
     }
   }
 
+  const users = participants
+    .map(p => ({fullName: stateProps._infoMap.getIn([p, 'fullname'], ''), username: p}))
+    .toArray()
   return {
     ...ownProps,
+    _generalChannelConversationIDKey,
+    _loadParticipants: dispatchProps._loadParticipants,
     conversationIDKey: stateProps.conversationIDKey,
     filter: stateProps._filter.toLowerCase(),
-    users: participants
-      .map(p => ({fullName: stateProps._infoMap.getIn([p, 'fullname'], ''), username: p}))
-      .toArray(),
+    loading: users.length === 0,
+    users,
   }
 }
 
-export default compose(connect(mapStateToProps, () => ({}), mergeProps), setDisplayName('UserMentionHud'))(
-  MentionHud
-)
+export default compose(
+  connect(mapStateToProps, mapDispatchToProps, mergeProps),
+  setDisplayName('UserMentionHud'),
+  lifecycle({
+    componentDidMount() {
```

Comment:
we should avoid using lifecycle in new stuff. really doesn't flow check at all

Reply:
OK, added a wrapper for it. I think this component's typing is kinda messed up in general, I'll ticket taking a look at it.